app/models/MovementValidationFailure.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.libs.json.{Format, Json}
+
+case class MovementValidationFailure(
+                                      errorType: Int,
+                                      errorReason: String
```

Comment:
Have we got it confirmed that errorType and errorReason are mandatory within EIS?

Reply:
It's a bonus if they are always defined but I suppose if they weren't defined, we wouldn't show it. However, if it were that all errors did not have a reason, we'd end up showing an empty notification banner, unless we guard against that?

I'd say make it optional but only if it does not result in a huge logic change.